src/node_file.cc

```diff
@@ -433,6 +435,70 @@ Local<Value> BuildStatsObject(Environment* env, const uv_stat_t* s) {
   return handle_scope.Escape(stats);
 }
 
+// Used to speed up module loading.  Returns the contents of the file as
+// a string or undefined when the file cannot be opened.  The speedup
+// comes from not creating Error objects on failure.
+static void InternalModuleReadFile(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsString());
+  node::Utf8Value path(env->isolate(), args[0]);
+
+  FILE* const stream = fopen(*path, "rb");
+  if (stream == nullptr) {
+    return;
+  }
+
+  std::vector<char> chars;
+  for (;;) {
+    const size_t kBlockSize = 32 << 10;
+    const size_t start = chars.size();
+    chars.resize(start + kBlockSize);
+    const size_t numchars = fread(&chars[start], 1, kBlockSize, stream);
+    if (numchars < kBlockSize) {
+      chars.resize(start + numchars);
+    }
+    if (numchars == 0) {
+      break;
+    }
+  }
+
+  fclose(stream);
```

Comment:
@bnoordhuis
And if the fs gets unmounted or the connection the the remote fs gets lost?